internal/nix/input.go

```diff
@@ -407,6 +415,40 @@ func (p *Package) hashFromNixPkgsURL() string {
 	return HashFromNixPkgsURL(p.URLForFlakeInput())
 }
 
+// BinaryCacheStore is the store from which to fetch this package's binaries.
+// It is used as FromStore in builtins.fetchClosure.
+const BinaryCacheStore = "https://cache.nixos.org"
+
+func (p *Package) IsInBinaryStore() bool {
+	return p.isVersioned()
+}
```

Comment:
Is there a better way to determine this? For example only true if lock file data has system info?

Reply:
🤔 yeah, we could add that check too. In my previous iteraiton of this PR, I was checking both `p.isVersioned()` and "has system info". 

I can add that here. It feels weird to possibly make network-calls in an if-statement but I understand the architecture better now so lets roll with it.